server/report.go

```diff
@@ -88,17 +88,26 @@ func reportForm(w http.ResponseWriter, r *http.Request) {
 
 // Render a list of reports for the board
 func reportList(w http.ResponseWriter, r *http.Request) {
+	var rep []auth.Report
+	var err error
 	board := extractParam(r, "board")
-	if !auth.IsNonMetaBoard(board) {
+
+	if !auth.IsBoard(board) {
 		text404(w)
 		return
 	}
 
-	rep, err := db.GetReports(board)
+	if board == "all" {
+		rep, err = db.GetReports(board, true)
+	} else {
+		rep, err = db.GetReports(board, false)
+	}
+
```

Comment:
`rep, err := db.GetReports(board, board == 'all')`

Reply:
That's the power of sleep deprivation for you!